packages/github/src/github.ts

```diff
@@ -119,16 +125,31 @@ export class GitHub extends Octokit {
   }
 
   private static getProxyAgent(
+    destinationUrl: string,
     options: Octokit.Options
   ): http.Agent | undefined {
     if (!options.request?.agent) {
-      const serverUrl = 'https://api.github.com'
-      if (httpClient.getProxyUrl(serverUrl)) {
+      if (httpClient.getProxyUrl(destinationUrl)) {
         const hc = new httpClient.HttpClient()
-        return hc.getAgent(serverUrl)
+        return hc.getAgent(destinationUrl)
       }
     }
 
     return undefined
   }
+
+  private static getApiBaseUrl(): string {
+    return process.env['GITHUB_API_URL'] || 'https://api.github.com'
+  }
+
+  private static getGraphQLBaseUrl(): string {
+    let url =
+      process.env['GITHUB_GRAPHQL_URL'] || 'https://api.github.com/graphql'
+
+    // Remove trailing "/graphql"
+    if (url.toUpperCase().endsWith('/GRAPHQL')) {
```

Comment:
There's no possibility of a trailing slash is there?  Alternative https://stackoverflow.com/a/16750711/775184

Reply:
shouldnt be a trailing slash, i'll trim on launch side when setting the urls

Reply:
also not bad to add resiliency here, i added code to trim trailing slash if detected